### PR TITLE
top-level: Make cross compiling slightly saner

### PR DESCRIPTION
**Pull request by Ericson2314**
###### Motivation for this change

Removes the weird stdenv cycle used to match the old infrastructure.
It turns out that matching it so precisely is not needed.

CC @nbp 

###### Things done

Tested with https://gist.github.com/Ericson2314/fc745290db68b20235bdc457961b3658, try:
```
nix-instantiate --eval -E 'import (builtins.fetchTarball https://gist.github.com/Ericson2314/fc745290db68b20235bdc457961b3658/archive/master.tar.gz) { me = "184adcc98c27bb75ad7f59c09dd93bfd978c24e6"; base = "a24728fe295f28648f1812b3565541f7ef6269f1"; }' -A testsPass
```

### CODE REVIEW

**Comment by nbp** [comment]
@Ericson2314 what is the error reported by Travis? It sounds like a legitimate failure.

**Comment by Ericson2314** [comment]
@nbp Ah, ran into this long ago. This should fix it. OK to merge if it does? (I have push access now.)

**Review by nbp — APPROVED** [review]

**Pull request merged**
